integration-tests/main/src/test/java/io/quarkus/it/main/HealthCheckTestCase.java

```diff
@@ -22,7 +21,6 @@
 
     @Test
     public void testInjection() {
-        Assumptions.assumeFalse(getClass().isAnnotationPresent(SubstrateTest.class));
```

Comment:
I don't know the reason, but this test was designed to prevent it from being launch in native mode. That's why the presence of both `@SubstrateTest` and `@NativeImageTest` is checked here. It should probably stay that way until `@SubstrateTest` is removed.

Reply:
The reason is because injection is not supported in native mode. I think we can safely remove Substrate here, if it is launched as native mode, the test will fail.

Reply:
That's good to know, but this reason might not be obvious to everyone. I'd rather keep these two checks and add an explicit message in case of failure. And also a comment explaining why we're testing `@SubstrateTest` :) That way, we'd be sure nobody will ever try to run this test in native mode even if the injection is failing.

Reply:
Actually, we should probably use `@DisabledOnNativeImage` instead.

Reply:
It is minor but I still think we can remove the `@SubstrateTest` check here: I do not think we'll try to run this test in native mode with `@SubstrateTest` - since it is deprecated. 

- I can however add a comment - 
```
The Injection test is not meant to be ran in native mode as it is not supported - https://quarkus.io/guides/getting-started-testing#native-executable-testing
```